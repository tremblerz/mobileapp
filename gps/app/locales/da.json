{
  "_display_name": "Dansk",
  "about": {
    "dimensions": "Skærmopløsning:",
    "operating_system_abbr": "OS:",
    "version": "Version:"
  },
  "common": {
    "done": "Udført"
  },
  "history": {
    "no_exposure": "Ingen kendte",
    "possible_exposure": "Mulig",
    "possible_exposure_para": "Det er muligt, at du har været i kontakt med eller tæt på nogen, som er testet positiv for COVID-19.",
    "what_does_this_mean": "Hvad betyder det?",
    "what_does_this_mean_para": "Baseret på din placeringshistorik er det muligt, at du har været i kontakt med eller tæt på nogen, som er diagnosticeret med COVID-19. Det behøver ikke at betyde, at du selv er smittet, men at du kan være det.\n\nFor yderligere information om hvad du skal gøre, så kig på myndighedernes webside om Coronavirus/COVID-19 i Danmark: https://coronasmitte.dk",
    "what_if_no_symptoms": "Hvad hvis jeg ikke har nogen symptomer?",
    "what_if_no_symptoms_para": "Hvis du ikke har nogen symptomer, men alligevel gerne vil testes, så ring til din læge.\n\nPersoner, som ikke har nogen symptomer, kan alligevel godt være smittede og smitte andre. Det er vigtigt, at du er forsigtig med at holde afstand til andre og lader være med at komme i kontakt med større grupper og personer i risikogrupperne (fx ældre og personer med kroniske lidelser) for at beskytte dig selv og andre."
  },
  "import": {
    "button_text": "Importér tidligere placeringer",
    "error": "Der gik noget galt med at importere dine data.",
    "google": {
      "already_imported": "Den valgte Takeout-fil er allerede importeret.",
      "disclaimer": "Safe Paths har ingen tilknytning til Google og deler aldrig dine data.",
      "file_open_error": "Filen kunne ikke åbnes. \nSørg for, at filen åbnes fra Google Drev.",
      "instructions_detailed": "Besøg Google Takeout og eksportér din placeringshistorik med følgende indstillinger:\n1. Leveringsmetode: \"Tilføj til Drev\"\n2. Hyppighed: \"Eksporter én gang\"\n3. Filtype & -størrelse: \".zip\" og \"1GB\"\n4. Google sender en email til dig, når eksporten er klar\n5. Kom tilbage hertil for at importere placeringer fra Google Drev",
      "instructions_first": "Ved at tilføje placeringsdata fra Google får du et forspring med at opbygge data, om hvor du har været for nyligt.",
      "instructions_second": "Før du kan importere, skal du først hente dine placeringsdata fra Google Takeout.",
      "invalid_file_format": "Det valgte filformat er ikke understøttet.\nUnderstøttede filformater: \".zip\".",
      "no_recent_locations": "Takeout har ikke nogen nylige placeringsdata.",
      "title": "Google Maps",
      "visit_button_text": "Besøg Google Takeout"
    },
    "subtitle": "For at se om du har været i nærheden af nogen med COVID-19, før du hentede denne app, kan du importere din personlige placeringshistorik.",
    "success": "Det lykkedes at importere dine nylige placeringer!",
    "title": "Importér placeringshistorik"
  },
  "label": {
    "about_title": "Om",
    "authorities_add_button_label": "Tilføj betroet kilde",
    "authorities_add_url": "Tilføj myndighed via URL",
    "authorities_desc": "Vælg en eller flere sundhedsmyndigheder, du stoler på, i dit område for at få data om mulig udsættelse for COVID-19. Vælg enten et navn fra den globale oversigt eller angiv webadressen, som er stillet til rådighed af en myndighed, der har implementeret Safe Paths.",
    "authorities_input_placeholder": "Indsæt din URL her",
    "authorities_new_in_area_msg": "Tilmeld dig til {{count}} ny betroet sundhedsmyndighed i dit område",
    "authorities_new_in_area_title": "Ny sundhedsmyndighed",
    "authorities_new_in_area_title_plural": "Nye sundhedsmyndigheder",
    "authorities_new_subcription_msg": "Du er tilmeldt til {{count}} ny myndighed i dit område",
    "authorities_new_subcription_msg_plural": "Du er tilmeldt til {{count}} nye myndigheder i dit område",
    "authorities_new_subcription_title": "{{count}} ny tilmelding",
    "authorities_new_subcription_title_plural": "{{count}} nye tilmeldinger",
    "authorities_new_in_area_msg_plural": "Tilmeld dig til {{count}} nye betroede sundhedsmyndigheder i dit område",
    "authorities_no_sources": "Ingen datakilde valgt endnu",
    "authorities_removal_alert_cancel": "Annullér",
    "authorities_removal_alert_desc": "Er du sikker på, at du vil fjerne denne myndighedsdatakilde?",
    "authorities_removal_alert_proceed": "Fortsæt",
    "authorities_removal_alert_title": "Fjern myndighed",
    "authorities_title": "Betroede kilder",
    "auto_subscribe_checkbox": "Aktivér automatisk tilmelding",
    "choose_provider_subtitle": "For at blive informeret om udsættelse for smitte er du nødt til at tilmelde dig en sundhedsmyndighed.",
    "choose_provider_title": "Vælg sundhedsmyndighed",
    "commitment": "Vi lover",
    "commitment_para": "Safe Paths optager og undersøger din kontakt med andre mennesker ved brug af din placering på en sikker måde. Dine data forlader ALDRIG din telefon uden dit samtykke.",
    "default_news_site_name": "Nyheder om Safe Paths",
    "enter_authority_url": "Indtast eller indsæt URL",
    "event_history_subtitle": "Forstå din egen eksponering for smitte ud fra data, som bliver delt med dig af sundhedsmyndigheder.",
    "event_history_title": "Eksponeringshistorik",
    "filter_authorities_by_gps_history": "Filtrér efter placering",
    "home_at_risk_header": "Du har måske været udsat for smitte",
    "home_at_risk_subsubtext": "Det betyder ikke nødvendigvis, at du selv er smittet.",
    "home_at_risk_subtext": "Baseret på din placeringshistorik er det muligt, at du har været i kontakt med eller tæt på nogen, som er diagnosticeret med COVID-19.",
    "home_enable_location": "Aktivér placeringsdata",
    "home_mayo_link_heading": "Mere information om COVID-19",
    "home_mayo_link_label": "fra Mayo Clinic",
    "home_no_contact_header": "Ingen kendt kontakt",
    "home_no_contact_subtext": "Baseret på tilgængelige data har du ikke være i nærheden af nogen, som er testet positive med COVID-19.",
    "home_setting_off_header": "Ukendt",
    "home_setting_off_subtext": "Vi kan ikke fortælle dig, om du har været i nærheden af smittede personer, medmindre du aktiverer placeringshistorik i indstillingerne.",
    "home_unknown_header": "Ukendt",
    "home_unknown_subtext": "Vi kan ikke fortælle dig, om du har været i nærheden af smittede personer, medmindre du giver applikationen adgang til din placering.",
    "latest_news": "Seneste nyheder",
    "launch_authority_access": "Tilmeld dig sundhedsmyndigheder i dit område",
    "launch_authority_header": "Sundhedsmyndigheder vil levere lokale data til din enhed, så du kan få at vide, om du har været i nærheden af en person, som er smittet.",
    "launch_authority_subheader": "Meld dig automatisk til at modtage de seneste opdateringer fra sundhedsmyndigheder i dit område.",
    "launch_done_header": "Alt udført",
    "launch_done_subheader": "Nu er du klar. Husk, at du altid kan ændre dine indstillinger senere.",
    "launch_enable_auto_subscription": "Aktivér automatisk tilmelding",
    "launch_enable_location": "Aktivér placering",
    "launch_enable_notif": "Aktivér notifikationer",
    "launch_finish_set_up": "Afslut opsætning",
    "launch_get_started": "Lad os komme i gang",
    "launch_location_access": "Adgang til placering",
    "launch_location_header": "For at huske, hvor du har været, har din telefon brug for at gemme din placering.",
    "launch_location_subheader": "Bare rolig. Informationerne forlader aldrig din enhed, medmindre du selv vælger at dele dem.",
    "launch_next": "Næste",
    "launch_notif_header": "Notifikationer fortæller dig, om du har været i nærheden af en smittet person.",
    "launch_notif_subheader": "Vi vil ikke genere dig med notifikationer, medmindre det er for at fortælle dig om mulig udsættelser for smitte.",
    "launch_notification_access": "Tillad notifikationer",
    "launch_screen1_header": "Vejen tilbage til en normal tilværelse begynder her.",
    "launch_screen2_header": "Få besked, hvis du har været i nærheden af nogen, som senere bliver diagnosticeret med COVID-19.",
    "launch_screen2_subheader": "Viden er vores styrke.",
    "launch_screen3_header": "Hvis du selv bliver testet positiv, kan du vælge at donere dine data anonymt",
    "launch_screen3_subheader": "Det hjælper med til at holde vores samfund sikkert.",
    "launch_screen4_header": "Du har fuld kontrol. Data bliver udelukkende gemt på din telefon.",
    "launch_screen4_subheader": "Hvis du bliver testet positiv, så er det kun dig der kan beslutte om du vil dele den viden.",
    "launch_set_up_phone": "Gør min telefon klar",
    "legal_page_title": "Juridisk",
    "loading_public_data": "henter data…",
    "location_disabled_message": "COVID Safe Paths kræver placeringstjenester.",
    "location_disabled_title": "Placeringssporing blev slået fra",
    "location_enabled_message": "COVID Safe Paths gemmer på sikker vis dine GPS koordinater på din enhed hvert femte minut.",
    "location_enabled_title": "COVID Safe Paths er slået til",
    "logging_active": "Placering aktiv",
    "logging_inactive": "Placering inaktiv",
    "news_subtitle": "Læs de seneste opdateringer om COVID-19 fra dine sundhedsmyndigheder og fra andre kilder.",
    "news_title": "Seneste nyheder",
    "no_data": "Ingen data",
    "push_at_risk_message": "Du har været i nærheden af en COVID-19-patient",
    "push_at_risk_title": "Du er muligvis i risiko",
    "see_exposure_history": "Se eksponeringshistorik",
    "settings_title": "Overblik",
    "skip_this_step": "Spring dette trin over",
    "team": "Holdet",
    "team_para": "Holdet bag denne applikation består af et konsortium af epidemiologer, ingeniører, dataloger, privatlivseksperter, professorer og forskere fra velrenommerede institutioner herunder: MIT, Harvard, The Mayo Clinic, TripleBlind, EyeNetra, Ernst & Young og Link Ventures.",
    "terms_of_use": "Vilkår for brug"
  },
  "onboarding": {
    "eula_checkbox": "Jeg accepterer licensaftalen",
    "eula_continue": "Fortsæt",
    "eula_message": "*Du skal acceptere for at kunne bruge Safe Paths"
  },
  "share": {
    "button_text": "Del placeringsdata",
    "paragraph_first": "Hvis du bliver testet positiv for COVID-19, så vær venlig at hjælpe dine lokale sundhedsmyndigheder ved at dele din placeringshistorik.",
    "paragraph_second": "Dine placeringer vil blive delt som en simpel liste bestående af tidspunkter og steder og intet andet.",
    "subtitle": "Dine private data kan overføres til sundhedsmyndigheder, sikkerhedskopieres eller på anden vis deles.",
    "title": "Del placeringshistorik"
  },
  "version_update": {
    "alert_label": "COVID Safe Paths er for gammel",
    "alert_sublabel": "Opdatér til en ny version?",
    "later": "Senere",
    "push_notification_message": "Opdatér venligst din applikation",
    "push_notification_title": "Applikationen er for gammel",
    "update": "Opdatér"
  }
}
