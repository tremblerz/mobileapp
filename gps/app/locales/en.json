{
  "_display_name": "English",
  "about": {
    "dimensions": "Dimensions:",
    "operating_system_abbr": "OS:",
    "version": "Version:"
  },
  "authorities": {
    "invalid_url_body": "The URL you entered was invalid, please try again",
    "invalid_url_title": "Invalid URL"
  },
  "common": {
    "done": "Done",
    "ok": "Ok"
  },
  "history": {
    "no_exposure": "No known",
    "possible_exposure": "Possible",
    "possible_exposure_para": "It is possible you were in contact with or close to someone who tested positive for COVID-19",
    "what_does_this_mean": "What does this mean?",
    "what_does_this_mean_para": "Based on your GPS history, it is possible that you may have been in contact with or close to somebody who was diagnosed with COVID-19. This does not mean you are infected but that you might be.\n\nFor further information on what you should do you can refer to the Mayo Clinic’s website.",
    "what_if_no_symptoms": "What if I’m not showing symptoms?",
    "what_if_no_symptoms_para": "If you have no symptoms but still would like to be tested you can go to your nearest testing site.\n\nIndividuals who don't exhibit symptoms can sometimes still carry the infection and infect others. Being careful about social distancing and coming in contact with large groups or at risk individuals (the elderly, those with significant other medical issues) is important to manage both your risk and the risk to others."
  },
  "import": {
    "button_text": "Import past locations",
    "error": "Something went wrong while importing your data.",
    "google": {
      "already_imported": "Provided Takeout file has already been imported.",
      "disclaimer": "Safe Paths has no affiliation with Google and never shares your data.",
      "file_open_error": "Could not open the file. \nPlease, make sure the file is opened from Google Drive",
      "instructions_detailed": "Visit Google Takeout and export your Location History using the following settings: \n1. Delivery method: \"Add to Drive\" \n2. Frequency: \"Export once\" \n3. File type & size: \".zip\" and \"1GB\"\n4. Google sends an email when the export is ready \n5. Return here to import locations from Google Drive",
      "instructions_first": "Adding location data from Google will give you a head start on building your recent locations.",
      "instructions_second": "Before you can import, you must first \"Take out\" your location data from Google.",
      "invalid_file_format": "Provided file format is not supported. \nSupported formats: \".zip\".",
      "no_recent_locations": "Takeout doesn't have any recent locations.",
      "title": "Google Maps",
      "visit_button_text": "Visit Google Takeout"
    },
    "subtitle": "To see if you encountered someone with COVID-19 prior to downloading this app, you can import your personal location history.",
    "success": "Recent locations has been successfully imported!",
    "title": "Import Locations"
  },
  "label": {
    "about_title": "About",
    "authorities_add_button_label": "Add Trusted Source",
    "authorities_add_url": "Add authority via URL",
    "authorities_desc": "Choose trusted healthcare authorities in your area to obtain exposure data. Either select a name from the global registry, or enter the web address provided by an authority which has implemented Safe Paths.",
    "authorities_input_placeholder": "Paste your URL here",
    "authorities_new_in_area_msg": "Subscribe to {{count}} new trusted Healthcare Authority in your location",
    "authorities_new_in_area_msg_plural": "Subscribe to {{count}} new trusted Healthcare Authorities in your location",
    "authorities_new_in_area_title": "New Healthcare Authority",
    "authorities_new_in_area_title_plural": "New Healthcare Authorities",
    "authorities_new_subcription_msg": "You have been subscribed to {{count}} new authority in your location",
    "authorities_new_subcription_msg_plural": "You have been subscribed to {{count}} new authorities in your location",
    "authorities_new_subcription_title": "{{count}} New Subscription",
    "authorities_new_subcription_title_plural": "{{count}} New Subscriptions",
    "authorities_new_subscription_msg": "You have been subscribed to {{count}} new authority in your location",
    "authorities_new_subscription_msg_plural": "You have been subscribed to {{count}} new authorities in your location",
    "authorities_new_subscription_title": "{{count}} New Subscription",
    "authorities_new_subscription_title_plural": "{{count}} New Subscriptions",
    "authorities_no_sources": "No data source yet",
    "authorities_removal_alert_cancel": "Cancel",
    "authorities_removal_alert_desc": "Are you sure you want to remove this authority data source?",
    "authorities_removal_alert_proceed": "Proceed",
    "authorities_removal_alert_title": "Remove Authority",
    "authorities_title": "Trusted Sources",
    "auto_subscribe_checkbox": "Enable auto subscription",
    "choose_provider_subtitle": "To be informed of exposures you will need to subscribe to a Health Authority.",
    "choose_provider_title": "Choose Health Authority",
    "commitment": "Commitment",
    "commitment_para": "Safe Paths is a global movement to develop free, open-source, privacy-by-design tools for residents, public health officials, and larger communities to flatten the curve of COVID-19, reduce fear, and prevent a surveillance-state response to the pandemic.\n\nFor more information, please visit ",
    "default_news_site_name": "Safe Paths News",
    "enter_authority_url": "Enter or paste URL",
    "event_history_subtitle": "Understand your personal exposure based on information shared by health authorities.",
    "event_history_title": "Exposure history",
    "filter_authorities_by_gps_history": "Filter by your locations",
    "home_at_risk_header": "You May Be Exposed",
    "home_at_risk_subsubtext": "This does not mean you are infected.",
    "home_at_risk_subtext": "Based on your GPS history, it is possible you were in contact with or close to someone diagnosed with COVID-19.",
    "home_enable_location": "Enable Location Data",
    "home_mayo_link_heading": "More COVID-19 information",
    "home_mayo_link_label": "from the Mayo Clinic",
    "home_no_contact_header": "No known contact",
    "home_no_contact_subtext": "Based on available data you haven’t been near anyone reported positive for COVID-19.",
    "home_setting_off_header": "Unknown",
    "home_setting_off_subtext": "We can’t tell if you’re at risk unless you enable location history in the settings screen.",
    "home_unknown_header": "Unknown",
    "home_unknown_subtext": "We can’t tell if you’re at risk unless you enable the app to access your location.",
    "import_success": "Imported successfully",
    "latest_news": "Latest News",
    "launch_authority_access": "Subscribe to nearby Health Authorities",
    "launch_authority_header": "Healthcare Authorities will provide your device with the local data to know if you have crossed paths with an infected person",
    "launch_authority_subheader": "Automatically subscribe to receive the latest updates from Healthcare Authorities in your area.",
    "launch_done_header": "All finished",
    "launch_done_subheader": "You’re ready to roll. Remember, you can always update your preferences later.",
    "launch_enable_auto_subscription": "Enable auto subscription",
    "launch_enable_location": "Enable Location",
    "launch_enable_notif": "Enable Notifications",
    "launch_finish_set_up": "Finish Setup",
    "launch_get_started": "Get Started",
    "launch_location_access": "Location access",
    "launch_location_header": "To remember where you go, your phone needs to save your location.",
    "launch_location_subheader": "Don’t worry, information never leaves your device unless you explicitly decide to share.",
    "launch_next": "Next",
    "launch_notif_header": "Notifications will let you know if you cross paths with an infected person.",
    "launch_notif_subheader": "We won't bother you except to share updates on your potential exposure risks.",
    "launch_notification_access": "Allow notifications",
    "launch_screen1_header": "The way back to normal starts here.",
    "launch_screen2_header": "Get notified if you cross paths with someone later diagnosed for COVID-19.",
    "launch_screen2_subheader": "Knowledge is power.",
    "launch_screen3_header": "If you test positive, you can choose to donate your data anonymously",
    "launch_screen3_subheader": "Which helps keep your entire community safe.",
    "launch_screen4_header": "You're in complete control. Data is only saved on your phone.",
    "launch_screen4_subheader": "If you test positive, you alone can choose whether to share.",
    "launch_set_up_phone": "Set up my phone",
    "legal_page_title": "Legal",
    "loading_public_data": "loading data...",
    "location_disabled_message": "COVID Safe Paths requires location services.",
    "location_disabled_title": "Location Tracking Was Disabled",
    "location_enabled_message": "COVID Safe Paths is securely storing your GPS coordinates once every five minutes on this device.",
    "location_enabled_title": "COVID Safe Paths Enabled",
    "logging_active": "Location Active",
    "logging_inactive": "Location Inactive",
    "news_subtitle": "Read about the latest COVID updates from your health authority and in general.",
    "news_title": "Latest news",
    "no_data": "No data",
    "privacy_policy": "Privacy Policy",
    "push_at_risk_message": "You have crossed paths with a COVID-19 patient",
    "push_at_risk_title": "You may be at risk",
    "see_exposure_history": "See exposure history",
    "settings_title": "Dashboard",
    "skip_this_step": "Skip this step",
    "team": "Team",
    "team_para": "Our team is composed of a consortium of epidemiologists, engineers, data scientists, digital privacy evangelists, professors and researchers from reputable institutions, including: MIT, Harvard, The Mayo Clinic, TripleBlind, EyeNetra, Ernst & Young and Link Ventures.",
    "terms_of_use": "Terms of use"
  },
  "onboarding": {
    "eula_checkbox": "I accept the licensing agreement",
    "eula_continue": "Continue",
    "eula_message": "*You must accept in order to use Safe Paths"
  },
  "share": {
    "button_text": "Share location data",
    "paragraph_first": "If you test positive for COVID-19, please do your part by sharing your location history with local authorities.",
    "paragraph_second": "Location is shared as a simple list of times and places, no additional information.",
    "subtitle": "Your private data can be transferred to health authorities, backed up, or otherwise shared.",
    "title": "Share location history"
  },
  "version_update": {
    "alert_label": "COVID Safe Paths is outdated",
    "alert_sublabel": "Update to a new version?",
    "later": "Later",
    "push_notification_message": "Please update your application",
    "push_notification_title": "Application is outdated",
    "update": "Update"
  }
}
