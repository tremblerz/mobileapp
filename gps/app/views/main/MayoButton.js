import React from 'react';
import { useTranslation } from 'react-i18next';
import { Image, Linking, TouchableOpacity, View } from 'react-native';

import foreArrow from '../../../../shared/assets/images/foreArrow.png';
import { Typography } from '../../components/Typography';
import { styles } from './style';

const MAYO_COVID_URL = 'https://www.mayoclinic.org/coronavirus-covid-19';

export const MayoButton = () => {
  const { t, i18n } = useTranslation();
  const onPress = () => Linking.openURL(MAYO_COVID_URL);

  /**
   * Currently, the component is only enabled in English. This will
   * be updated once there are language-specific resources to link to.
   */
  const isEnabled = i18n.language === 'en';

  return (
    isEnabled && (
      <View>
        <TouchableOpacity onPress={onPress} style={styles.mayoInfoRow}>
          <View style={styles.mayoInfoContainer}>
            <Typography
              style={styles.mainMayoHeader}
              onPress={onPress}
              testID='MayoLinkHeading'>
              {t('label.home_mayo_link_heading')}
            </Typography>
            <Typography
              style={styles.mainMayoSubtext}
              onPress={onPress}
              testID='MayoLinkLabel'>
              {t('label.home_mayo_link_label')}
            </Typography>
          </View>
          <View style={styles.arrowContainer}>
            <Image source={foreArrow} />
          </View>
        </TouchableOpacity>
      </View>
    )
  );
};
