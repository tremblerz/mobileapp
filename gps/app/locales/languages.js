import './all-dayjs-locales';

import dayjs from 'dayjs';
import i18next from 'i18next';
import { initReactI18next } from 'react-i18next';
import { useTranslation } from 'react-i18next';
import { NativeModules, Platform } from 'react-native';

import { LANG_OVERRIDE } from '../constants/storage';
import { GetStoreData, SetStoreData } from '../helpers/General';
import ar from './ar.json';
import da from './da.json';
import en from './en.json';
import es from './es.json';
import fil from './fil.json';
import fr from './fr.json';
import ht from './ht.json';
import id from './id.json';
import it from './it.json';
import ja from './ja.json';
import ml from './ml.json';
import nl from './nl.json';
import pl from './pl.json';
import pt_BR from './pt_BR.json';
import ro from './ro.json';
import ru from './ru.json';
import sk from './sk.json';
import vi from './vi.json';
import zh_Hant from './zh_Hant.json';

// Refer this for checking the codes and creating new folders https://developer.chrome.com/webstore/i18n

// Adding/updating a language:
// 1. Add the language in Lokalise
// 2. run: yarn i18n:pull with your lokalise token, see gps/app/locales/pull.sh instructions
// 3. import xy from `./xy.json` and add the language to the language block

/** Fetch the user language override, if any */
export async function getUserLocaleOverride() {
  return await GetStoreData(LANG_OVERRIDE);
}

export function getLanguageFromLocale(locale) {
  const [languageCode] = toIETFLanguageTag(locale).split('-');
  return languageCode;
}

/**
 * Convert ISO language `en_US` to IETF `en-us` for dayjs
 *
 * @param {string} locale
 */
function toIETFLanguageTag(locale) {
  return locale.replace('_', '-').toLowerCase();
}

async function setLocale(locale) {
  dayjs.locale(toIETFLanguageTag(locale));
  return await i18next.changeLanguage(locale);
}

export function useLanguageDirection() {
  const { i18n } = useTranslation();
  return i18n.dir();
}

export async function setUserLocaleOverride(locale) {
  await setLocale(locale);
  await SetStoreData(LANG_OVERRIDE, locale);
}

/* eslint-disable no-underscore-dangle */
/** Languages only available in dev builds. */
const DEV_LANGUAGES = __DEV__
  ? {
    ar: { label: ar._display_name, translation: ar },
    da: { label: da._display_name, translation: da },
    es: { label: es._display_name, translation: es },
    fil: { label: fil._display_name, translation: fil },
    fr: { label: fr._display_name, translation: fr },
    id: { label: id._display_name, translation: id },
    it: { label: it._display_name, translation: it },
    ja: { label: ja._display_name, translation: ja },
    ml: { label: ml._display_name, translation: ml },
    nl: { label: nl._display_name, translation: nl },
    pl: { label: pl._display_name, translation: pl },
    pt_BR: { label: pt_BR._display_name, translation: pt_BR },
    ro: { label: ro._display_name, translation: ro },
    ru: { label: ru._display_name, translation: ru },
    sk: { label: sk._display_name, translation: sk },
    vi: { label: vi._display_name, translation: vi },
    zh_Hant: { label: zh_Hant._display_name, translation: zh_Hant },
  }
  : {};

i18next.use(initReactI18next).init({
  interpolation: {
    // React already does escaping
    escapeValue: false,
  },
  lng: 'en', // 'en' | 'es',
  fallbackLng: 'en', // If language detector fails
  returnEmptyString: false,
  resources: {
    en: { label: en._display_name, translation: en },
    ht: { label: ht._display_name, translation: ht },
    ...DEV_LANGUAGES,
  },
});
/* eslint-enable no-underscore-dangle */

/** The known locale list */
export const LOCALE_LIST = Object.entries(i18next.options.resources)
  .map(([langCode, lang]) => ({
    value: langCode,
    label: lang.label,
  }))
  .sort((a, b) => a.value > b.value);

/** A map of locale code to name. */
export const LOCALE_NAME = Object.entries(i18next.options.resources).reduce(
  (output, [langCode, lang]) => {
    output[langCode] = lang.label;
    return output;
  },
  {},
);

/** Get the device locale e.g. en_US */
export function getDeviceLocale() {
  return Platform.OS === 'ios'
    ? NativeModules.SettingsManager.settings.AppleLocale || // iOS < 13
    NativeModules.SettingsManager.settings.AppleLanguages[0] // iOS 13
    : NativeModules.I18nManager.localeIdentifier; // Android
}

/**
 * Find compatible supported i18n language
 *
 * e.g. device locale `en_AU` would find `en`
 *      device locale `pt_BR` would find `pt-BR`
 */
export function supportedDeviceLanguageOrEnglish() {
  const locale = getDeviceLocale(); // en_US
  const langCode = getLanguageFromLocale(locale); // en
  const found = Object.keys(LOCALE_NAME).find(
    l => l === langCode || toIETFLanguageTag(l) === toIETFLanguageTag(locale),
  );
  return found || 'en';
}

// detect and set device locale, must go after i18next.init()
setLocale(supportedDeviceLanguageOrEnglish());

// detect user override
getUserLocaleOverride().then(locale => locale && setLocale(locale));

export default i18next;

// do not remove, this will force the yarn i18n:extract to export this key
i18next.t('_display_name');
