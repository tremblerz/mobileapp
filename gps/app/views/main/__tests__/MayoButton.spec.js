import { act, fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import { Linking } from 'react-native';

import { setUserLocaleOverride } from '../../../locales/languages';
import { MayoButton } from '../MayoButton';

let mayoClinicURLSpy;

beforeEach(() => {
  mayoClinicURLSpy = jest
    .spyOn(Linking, 'openURL')
    .mockImplementation(() => Promise.resolve());
});

afterEach(() => {
  mayoClinicURLSpy.mockRestore();
});

it('does not render when the language is not english', async () => {
  await act(async () => {
    await setUserLocaleOverride('es');
  });

  const { asJSON } = render(<MayoButton />);

  expect(asJSON()).toMatchSnapshot();
});

it('renders when the language is english', async () => {
  await act(async () => {
    await setUserLocaleOverride('en');
  });

  const { asJSON } = render(<MayoButton />);

  expect(asJSON()).toMatchSnapshot();
});

it('open mayo clinic url with label', async () => {
  await act(async () => {
    await setUserLocaleOverride('en');
  });
  const { getByTestId } = render(<MayoButton />);

  const button = getByTestId('MayoLinkLabel');
  fireEvent.press(button);

  expect(mayoClinicURLSpy).toHaveBeenCalled();
});

it('open mayo clinic url with heading', async () => {
  await act(async () => {
    await setUserLocaleOverride('en');
  });
  const { getByTestId } = render(<MayoButton />);

  const button = getByTestId('MayoLinkHeading');
  fireEvent.press(button);

  expect(mayoClinicURLSpy).toHaveBeenCalled();
});
