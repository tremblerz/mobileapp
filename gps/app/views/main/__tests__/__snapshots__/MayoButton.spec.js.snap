// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render when the language is not english 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`renders when the language is english 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "flex-end",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#E5E4E6",
              "fontFamily": "IBMPlexSans-Bold",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24,
              "textAlign": "left",
              "writingDirection": "ltr",
            }
          }
          testID="MayoLinkHeading"
          use="body1"
        >
          More COVID-19 information
        </Text>
        <Text
          style={
            Object {
              "color": "#E5E4E6",
              "fontFamily": "IBMPlexSans",
              "fontSize": 18,
              "fontWeight": "normal",
              "lineHeight": 24,
              "textAlign": "left",
              "writingDirection": "ltr",
            }
          }
          testID="MayoLinkLabel"
          use="body1"
        >
          from the Mayo Clinic
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "paddingLeft": 20,
            "paddingRight": 20,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../shared/assets/images/foreArrow.png",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
